MTM-66671 Announcement regarding deprecation of tenant options in request headers (#4947)
SoftwareAG/c8y-docs: content/change-logs/platform-services/cumulocity-2026-230-0-no-tenant-options-as-headers-in-proxy.md

change_type:
  - value: change-inv-3bw8e
    label: Announcement 
component:
  - value: component-JlFdtOPva
    label: REST API
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-66671 
version: 2026.230.0
environment_availability:
---
The deprecation of attaching microservices' tenant options as request headers by the microservices proxy was under [Public Preview](/change-logs/?change-type=.change-type-announcement#cumulocity-2025-322-0-deprecation-of-tenant-options-in-request-headers), and is now Generally Available (GA) for CD versions 2026.230.0 and higher, and will be present in the 2027 annual release.

Until now, tenant options were attached to each microservice request unless this functionality was explicitly disabled via a feature toggle. The microservice proxy added the tenant options to the request headers and forwarded the request to the respective microservice. This functionality is now disabled by default. Tenant options are no longer attached as headers to requests to microservices. The retrieval of the tenant options remains possible through the endpoint <TENANT_DOMAIN>/application/currentApplication/settings.

> This change is now available by default. The old behaviour can still be enabled via the feature toggle `core.ms-proxy.no-tenant-options-in-headers`.

Reviewer: I think we should not actively point customers to the fact that they can still use the legacy behavior. It is good that we can restore it using the toggle for customers who run into problems. However we should not actively advertise this possibility.

```suggestion
This change is now available by default.
```

---

> date: '2026-07-21'

Reviewer 2: The date should be left empty as it is filled in automatically. The same for the environment availability. It will be added as soon as this version is available on the respective environments. Or did you add this intentionally as the feature is already available as preview feature?

Author: No, this was not intentional. I removed both.

title: Tenant options are not attached as request headers by microservices proxy 
product_area: Platform services
change_type:
  - value: change-inv-3bw8e
    label: Announcement 
component:
  - value: component-JlFdtOPva
    label: REST API
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-66671 
version: 2026.230.0
environment_availability:
  - label: eu-latest.cumulocity.com
  - label: apj.cumulocity.com
  - label: jp.cumulocity.com
  - label: emea.cumulocity.com
  - label: us.cumulocity.com
  - label: cumulocity.com
---

> The deprecation of attaching microservices' tenant options as request headers by the microservices proxy was under [Public Preview](/change-logs/?change-type=.change-type-announcement#cumulocity-2025-322-0-deprecation-of-tenant-options-in-request-headers), and is now to Generally Available (GA) for CD versions 2026.230.0 and higher, and will be present in the 2027 annual release.

Reviewer 2: ```suggestion
The deprecation of attaching microservices' tenant options as request headers by the microservices proxy was under [Public Preview](/change-logs/?change-type=.change-type-announcement#cumulocity-2025-322-0-deprecation-of-tenant-options-in-request-headers), and is now Generally Available (GA) for CD versions 2026.230.0 and higher, and will be present in the 2027 annual release.
```

> Until now, tenant options were attached to each microservice request unless this functionality was explicitly disabled via a feature toggle. The microservice proxy added the tenant options to the request headers and forwarded the request to the respective microservice. This functionality is now disabled per default. Tenant options are longer be attached as headers to requests to microservices. The retrieval of the tenant options remains possible through the endpoint <TENANT_DOMAIN>/application/currentApplication/settings.

Reviewer 2: ```suggestion
Until now, tenant options were attached to each microservice request unless this functionality was explicitly disabled via a feature toggle. The microservice proxy added the tenant options to the request headers and forwarded the request to the respective microservice. This functionality is now disabled by default. Tenant options can no longer be attached as headers to requests to microservices. The retrieval of the tenant options remains possible through the endpoint <TENANT_DOMAIN>/application/currentApplication/settings.
```

Author: @BeateRixen I used "by default", as proposed in the suggestion , but changed the broken sentence in a different way (not using "can no" , because there is a possibility to enable that again). Could you please check whether the change is fine with you?

Reviewer 2: Sounds good to me.

> This change is now available per default. The old behaviour can still be enabled via a feature toggle core.ms-proxy.no-tenant-options-in-headers.

Reviewer 2: ```suggestion
This change is now available by default. The old behaviour can still be enabled via the feature toggle `core.ms-proxy.no-tenant-options-in-headers`.
```
